### PR TITLE
Mobile support to show bnpl by code [29769]

### PR DESCRIPTION
**Pull request by lqgioan**
## Description
- Added methods(`showBNPLByCodes ) to show BNPL by codes
- Mark as deprecated methods(`getBNPLAvailability `, `showBNPL `)

## Motivation & Context
Ticket: https://app.shortcut.com/credify/story/29769/mobile-support-to-show-bnpl-by-code

## How has this been tested?
Test BNPL flow. The user can see offer detail and redeem that offer.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: Credify/Credify/serviceX.swift [hunk 1]
```diff
@@ -116,45 +117,93 @@ public struct serviceX {
...
+        }
+        
+        /// This kicks off offer redemption flow.
+        /// - Parameters:
+        ///   - from: ViewController that renders a new view from
+        ///   - offerCode: This is the offer code
+        ///   - userProfile: User object
+        ///   - pushClaimTokensTask: A task that calls your push claim token API. This SDK needs to receive success status of this task.
+        ///   - completionHandler: Completion handler. You can get notified about the result of the offer redemption flow.
+        public func presentOfferByCodeModally(
```
Comment (ngo275) [inline Credify/Credify/serviceX.swift:142]: `presentOfferModallyByCode`

File: Credify/Credify/serviceX.swift [hunk 2]
```diff
@@ -116,45 +117,93 @@ public struct serviceX {
         ///   - userProfile: User object
         ///   - pushClaimTokensTask: A task that calls your push claim token API. This SDK needs to receive success status of this task.
         ///   - completionHandler: Completion handler. You can get notified about the result of the offer redemption flow.
+        @available(*, deprecated, message: "Using presentByOfferCodeModally method instead")
```
Comment (ngo275) [inline Credify/Credify/serviceX.swift:120]: Please update the function name accordingly

Comment (lqgioan) [reply Credify/Credify/serviceX.swift:120]: @ngo275 I updated it

File: Credify/Credify/serviceX.swift (hunk 1 above)
Comment (lqgioan) [reply Credify/Credify/serviceX.swift:142]: @ngo275 I renamed it

**Review by ngo275 — APPROVED** [review]

**Pull request merged**